Change Assign Placenames robot to update cocina directly.

justinlittman opened the pull request:
closes #705

## Why was this change made? 🤔
Get rid of MODS.


## How was this change tested? 🤨

⚡ ⚠ If this change involves consuming from other services or writing to shared file systems, test that GIS accessioning works properly in [stage|qa] environment, in addition to specs. ⚡

Unit

justinlittman commented:
Note that this gets rid of the code that creates LC subject headings. It seems like this code would almost never be exercised, as there are very few possible cases when the existing subject is found in the gazetteer but the subject doesn't match the LC Subject.

Further note that it has been proposed to move to some sort of a dynamic lookup for the gazetteer instead of static data. See #732

edsu reviewed (changes requested):
Can you please say more about how you determined that "there are very few possible cases when the existing subject is found in the gazetteer but the subject doesn't match the LC Subject."?

edsu reviewed (changes requested):
If you are removing the LC functionality can the `Gazetteer.find_loc_*` methods and specs be removed. Also can the CSV be modified to no longer included this information that isn't used?

justinlittman commented:
@edsu I removed the loc methods from the gazetteer and added a comment about the legacy data. I chose not to change the data to avoid messing it up, especially given that we'll probably be replacing it later in the WC.
